### PR TITLE
Make Network instances reusable. Closes #469

### LINKED ISSUE #469: Network instances can't be used with ClassRule
From here:
https://stackoverflow.com/questions/46562926/no-such-network-error-when-trying-to-re-use-testcontainers-rules-with-network

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/Network.java [hunk 1]
```diff
@@ -42,10 +37,20 @@ static Network newNetwork() {
         private String driver;
 
         @Singular
-        private Set<Consumer<CreateNetworkCmd>> createNetworkCmdModifiers = new LinkedHashSet<>();
+        private Set<Consumer<CreateNetworkCmd>> createNetworkCmdModifiers;
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/Network.java]: Lombok's `@Builder` will initialize it to empty collection if `@Singular` is present

**Comment by rnorth** [comment]
This looks sensible to me. Please ignore the CircleCI failure - it's not set up correctly right now.

**Comment by bsideup** [comment]
@rnorth so... approved? :)

**Comment by rnorth** [comment]
@bsideup yes, approved, sorry!

**Comment by rnorth** [comment]
We still seem to have a merge conflict unfortunately...

**Pull request merged**